### PR TITLE
Remove references to Razor from RazorPageGenerator

### PR DESCRIPTION
**Pull request by pranavkm**
Resulting diff - https://github.com/aspnet/Diagnostics/commit/6a93fb2b65fd9ac84056a2a54e9dc30255745103

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/RazorEngineBuilderExtensions.cs [hunk 1]
```diff
@@ -57,6 +58,48 @@ public static IRazorEngineBuilder SetBaseType(this IRazorEngineBuilder builder,
...
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            ConfigureClass(builder, (document, @class) => @class.Name = className);
+            return builder;
+        }
+
+        public static IRazorEngineBuilder ConfigureClass(
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Razor.Evolution/RazorEngineBuilderExtensions.cs]: cc @rynowak

**Comment by pranavkm** [comment]
Expand the diff to see what's changed.

File: src/RazorPageGenerator/Program.cs [hunk 2]
```diff
@@ -23,7 +24,20 @@ public static void Main(string[] args)
...
+            var razorEngine = RazorEngine.Create(builder =>
+            {
+                builder
+                    .SetNamespace(rootNamespace)
+                    .SetBaseType("Microsoft.Extensions.RazorViews.BaseView")
+                    .ConfigureClass((document, @class) =>
+                    {
+                        @class.Name = Path.GetFileNameWithoutExtension(document.Source.Filename);
+                        @class.AccessModifier = "internal";
+                    });
```
Comment (rynowak) [inline src/RazorPageGenerator/Program.cs]: This is delicious. Exactly what I was hoping for.

File: src/RazorPageGenerator/Program.cs [hunk 3]
```diff
@@ -23,7 +24,20 @@ public static void Main(string[] args)
...
+
+                builder.Features.Add(new RemovePragamaChecksumFeature());
```
Comment (rynowak) [inline src/RazorPageGenerator/Program.cs]: typo `pragama`

File: src/RazorPageGenerator/Program.cs [hunk 4]
```diff
@@ -55,66 +72,104 @@ public static void Main(string[] args)
...
+            }
+
+            var generatedCodeFilePath = Path.ChangeExtension(
+                ((FileSystemRazorProjectItem)projectItem).FileInfo.FullName,
+                ".Designer.cs");
+            File.WriteAllText(generatedCodeFilePath, cSharpDocument.GeneratedCode);
         }
 
-        private static string ProcessFileIncludes(string basePath, string cshtmlContent)
+        private class FileSystemRazorProject : RazorProject
```
Comment (rynowak) [inline src/RazorPageGenerator/Program.cs]: This should live in Razor

File: src/Microsoft.AspNetCore.Razor.Evolution/RazorEngineBuilderExtensions.cs [hunk 5]
```diff
@@ -57,6 +58,48 @@ public static IRazorEngineBuilder SetBaseType(this IRazorEngineBuilder builder,
...
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            var configurationFeature = GetDefaultDocumentClassifierPassFeature(builder);
+            configurationFeature.ConfigureNamespace.Add((document, @namespace) => @namespace.Content = namespaceName);
+            return builder;
+        }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/RazorEngineBuilderExtensions.cs]: These new methods need tests

**Comment by pranavkm** [comment]
🆙 📅 @ajaybhargavb  \ @kichalla could you guys review this?

File: src/RazorPageGenerator/RemovePragmaChecksumFeature.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Microsoft.AspNetCore.Razor.Evolution;
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+
+namespace RazorPageGenerator
+{
+    class RemovePragmaChecksumFeature : RazorIRPassBase, IRazorIROptimizationPass
```
Comment (ajaybhargavb) [inline src/RazorPageGenerator/RemovePragmaChecksumFeature.cs]: public class

File: src/RazorPageGenerator/Program.cs [hunk 7]
```diff
@@ -4,8 +4,8 @@
 using System;
 using System.IO;
 using System.Linq;
-using Microsoft.AspNetCore.Razor;
-using Microsoft.AspNetCore.Razor.CodeGenerators;
+using System.Text;
+using Microsoft.AspNetCore.Razor.Evolution;
 
 namespace RazorPageGenerator
```
Comment (ajaybhargavb) [inline src/RazorPageGenerator/Program.cs]: Is it worth adding tests for `RazorPageGenerator`? I believe there are no tests for these right now. Not blocking this PR though.

File: src/RazorPageGenerator/RemovePragmaChecksumFeature.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
```
Comment (ajaybhargavb) [inline src/RazorPageGenerator/RemovePragmaChecksumFeature.cs]: Copyright

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/FileSystemRazorProjectTest.cs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        public void EnumerateItems_DiscoversAllCshtmlFiles()
+        {
+            // Arrange
+            var fileSystemProject = new FileSystemRazorProject(TestFolder);
+
+            // Act
+            var files = fileSystemProject.EnumerateItems("/");
+
+            // Assert
+            Assert.Collection(files,
```
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/FileSystemRazorProjectTest.cs]: This test is failing in linux. Looks like the ordering is different.

File: src/RazorPageGenerator/Program.cs (hunk 7 above)
Comment (pranavkm) [inline src/RazorPageGenerator/Program.cs]: Sure. Added a couple tests for it.

Comment (ajaybhargavb) [inline src/RazorPageGenerator/Program.cs]: Thanks. Looks good 👍

**Pull request merged**